src/libraries/System.Private.CoreLib/src/System/Threading/CancellationTokenSource.cs

```diff
@@ -922,28 +833,59 @@ protected override void Dispose(bool disposing)
             }
         }
 
-        internal sealed class CallbackPartition
+        private static void Invoke(Delegate d, object? state, CancellationTokenSource source)
         {
-            /// <summary>The associated source that owns this partition.</summary>
+            Debug.Assert(d is Action<object?> || d is Action<object?, CancellationToken>);
+
+            if (d is Action<object?> actionWithState)
+            {
+                actionWithState(state);
+            }
+            else
+            {
+                ((Action<object?, CancellationToken>)d)(state, new CancellationToken(source));
+            }
+        }
+
+        /// <summary>Set of all the registrations in the token source.</summary>
+        /// <remarks>
+        /// Separated out into a separate instance to keep CancellationTokenSource smaller for the case where one is created but nothing is registered with it.
+        /// This happens not infrequently, in particular when one is created for an operation that ends up completing synchronously / quickly.
+        /// </remarks>
+        internal sealed class Registrations
```

Comment:
Is there a perf benefit or something to sealing an internal class with no virtual methods? I see CallbackPartition was also sealed. I'm just curious.

Reply:
If there aren't any virtuals or interface implementations, no.  I'm just in the habit of sealing everything until reason dictates otherwise :-)

Reply:
[isinst is a bit faster for sealed classes](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGPKQAIyGBhBmgbxoZ6foeAQIAG1YALGGADWACgjAAVpIwN5CgJQMAvAD5VihgEtcDAGJCA3Nx7WedRoJHjJUgMoxswmABM5i5foa2npqRibunj7mEFbUvOxxvPZmQqx4GH5KYCpqmroMMtHqarHx8bZ8jBFe3tFpuBjVPpkBucEFTbVCxYqxAL40NMwpEJwVhgB2Kt4Arvj4AJ79g9S4HjVMpDyd0WOJPJPTc4vL1DRAA==)

Reply:
> isinst is a bit faster for sealed classes

Yup, for sealed types an `is` can then be achieved similar to GetType() == typeof(Target).